cogs/commands/quotes.py

```diff
@@ -0,0 +1,450 @@
+        db_session.commit()
+        return deleted
+    except (ScalarListException, SQLAlchemyError) as e:
+        db_session.rollback()
+        logging.exception(e)
+        raise QuoteException(QuoteError.DB_ERROR)
+
+
+def opt_in_to_quotes(requester: Mention, db_session=db_session) -> str:
+    # check to see if target is opted out already
+    if requester.is_id_type():
+        q = db_session.query(QuoteOptouts).filter(QuoteOptouts.user_id == requester.id)
+    else:
+        q = db_session.query(QuoteOptouts).filter(
+            QuoteOptouts.user_string == requester.string
+        )
+
+    if q.one_or_none() is None:
+        raise QuoteException(QuoteError.NO_OP)
+
+    # opt in
+    try:
+        q.delete()
+        db_session.commit()
+        return "OK"
+    except (ScalarListException, SQLAlchemyError) as e:
+        db_session.rollback()
+        logging.exception(e)
+        raise QuoteException(QuoteError.DB_ERROR)
+
+
+class QueryInputConverter(Converter):
+    async def convert(self, ctx, argument) -> Union[Mention, QuoteID, str]:
+        if is_id(argument):
+            return int(argument[1:])
+
+        if argument[0] == "@":
+            return Mention.string_mention(argument[1:])
+
+        argument = await MC.convert(ctx, argument)
+
+        if argument.is_id_type():
+            return argument
+
+        return argument
+
+
+class Quotes(commands.Cog):
+    def __init__(self, bot: Bot):
+        self.bot = bot
+
+    @commands.group(
+        invoke_without_command=True, help=LONG_HELP_TEXT, brief=SHORT_HELP_TEXT
+    )
+    async def quote(self, ctx: Context, *, query_arg: QueryInputConverter = None):
+        if query_arg is not None:
+            query = quotes_query(query_arg)
+        else:
+            query = db_session.query(Quote)
+
+        # select a random quote if one exists
+        q: Optional[Quote] = query.order_by(func.random()).first()
+
+        if q is None:
+            message = "No quote matched the criteria"
+        else:
+            # create message
+            message = quote_str(q)
+
+        # send message with no pings
+        await ctx.send(message, allowed_mentions=AllowedMentions().none())
+
+    @quote.command()
+    async def add(self, ctx: Context, author: MentionConverter, *, quote):
+        """Add a quote, format !quote add <author> <quote text>"""
+        requester = get_name_string(ctx)
+        now = datetime.now()
+
+        try:
+            quote_id = add_quote(author, quote, now)
+
+            result = f"Thank you {requester}, recorded quote with ID #{quote_id}."
+        except QuoteException as e:
+            if e.err == QuoteError.BAD_FORMAT:
+                result = "Invalid format: no quote to record."
+            elif e.err == QuoteError.OPTED_OUT:
+                result = "Invalid Author: User has opted out of being quoted."
+            elif e.err == QuoteError.DB_ERROR:
+                result = "Database error."
```

Comment:
Could you add a generic `else` branch to these except clauses please, just in case?